### ISSUE TITLE
DINO : Emerging Properties in Self-Supervised Vision Transformers

### ISSUE THREAD

**Issue by yskim0** [One-Day-One-Paper/One-Day-One-Paper#161]
# DINO : Emerging Properties in Self-Supervised Vision Transformers

### 3.1. SSL with Knowledge Distillation

- DINO → recent SSL + knowledge distillation
    - knowledge distillation is a learning paradigm where we train a student network $g_{\theta_s}$ to match the output of a given teacher network $g_{\theta_t}$
        - both network output probability distributions over K dimensions ⇒ normalizing the output of the network g with softmax operation + temperature
            
            ![스크린샷 2022-05-03 오후 4.48.33.png](https://s3-us-west-2.amazonaws.com/secure.notion-static.com/82885760-9269-4c13-a15d-c4f2ec66dbcd/스크린샷_2022-05-03_오후_4.48.33.png)
            
        - Given a fixted teacher network → learn to match these distributions by minimizing the CE loss w.r.t. the params of the student networks
            
            ![스크린샷 2022-05-03 오후 4.50.45.png](https://s3-us-west-2.amazonaws.com/secure.notion-static.com/df1be67e-ec83-4f0b-81f6-2e466b1d3ce4/스크린샷_2022-05-03_오후_4.50.45.png)
            
        
    - **[중요] how we adapt the problem in Eq(2) to self-supervised learning**
        - First, construct different distorted views, crops, multi-crop... ⇒ **generate a set V of different views**
            - this set contains two **global** views ($x^g_1$ and $x^g_2$) and several **local** views of smaller resolution
            - all crops are passed → student
            only the global views are passed → teacher
                - “local-to-global” correspondences
            
            minimize the loss w.r.t. $\theta_s$
            
            ![스크린샷 2022-05-03 오후 4.49.43.png](https://s3-us-west-2.amazonaws.com/secure.notion-static.com/35d6adac-10c3-4e26-b961-4ef0acd6c1ff/스크린샷_2022-05-03_오후_4.49.43.png)
            
        - **teacher network**
            - Unlike KD, we do not have a teacher $g_{\theta_t}$ ⇒ we build it from **past iterations** of the student network
            - ***freezing the teacher network over an epoch*** works well in the framework
            ↔ copying the student weight for the teacher fails to converge
            - **EMA** on the sutdent weighter (**== momentum encoder**)
            $\theta_t \leftarrow \lambda\theta_t + (1-\lambda)\theta_s$
                - MoCo에서는 contrastive learning 을 위한 거였지만 여기서는 **self-training을 사용하기 위한 *mean teacher로 사용***
                - 이 teacher는 모델 앙상블처럼 보여지기도... ]
                - *Polyak-Ruppert averaging?*
            - 어쨌든 이 teacher는 guide the training of the sutdnet by providing target features of higher qualities.
        - **network architecture**
            - do not use a predictor, resulting ni the exact same archi. in student and teacher
        - **avoiding collapse**
            - while our framework can be stabilized with multiple normalizations
            - also work with **only a centering and sharpening of the momentum `teacher`** outputs to avoid model collapse
                - centering → encourage uniform dist.
                - sharpening → opposite effect
                - 위 두 개를 적절히 사용하면 avoid collapse in presence of a momentum teacher
            - centering **(about teacher!!)**
                - only depends on first-order batch statistics and can be interpreted as adding a bias term c to the teacher
                $g_t(x) \leftarrow g_t(x) + c$
                - center c is updated with an EMA, which allows the approach to work well across different batch sizes
                $c \leftarrow mc + (1-m)\frac 1 B \sum^B_{i=1} g_{\theta_t}(x_i)$
            - sharpening
                - 위에 teacher network softmax normalization할 때 temperature 낮게 주면 됨
